docs/CIRCUIT_JSON_PCB_OVERVIEW.md

```diff
@@ -134,7 +134,24 @@ export interface PcbPlatedHoleOval {
   pcb_plated_hole_id: string
 }
 
-export type PcbPlatedHole = PcbPlatedHoleCircle | PcbPlatedHoleOval
+export interface PcbHoleCircularWithSquarePlated {
+  type: "pcb_plated_hole"
+  shape: "circular_hole_with_square_pad"
+  hole_shape: "circle"
+  pad_shape: "square"
+  hole_diameter: number
+  square_pad_width: number
+  square_pad_height: number
+  x: Distance
+  y: Distance
+  layers: LayerRef[]
+  port_hints?: string[]
+  pcb_component_id?: string
+  pcb_port_id?: string
+  pcb_plated_hole_id: string
```

Comment:
sorry idk maybe i gave you the wrong idea but this is the props repo, so everything should be camelcase

CC @Abse2001 we gotta integrate tscircuit-lint on this repo or somehow make this check automatic!

`props` uses `camelCase`

`circuit-json` uses `snake_case`

Sorry if i said the wrong thing earlier because i was confused- the circuit-json repo needs a lot of the properties you mentioned here. I do a lot of reviews switching repos so I do something forget which repo i'm reviewing